File: solutions/problem-10/sumofprimes.c
#include<stdio.h>

int main(){

    int i,k,f,n,sum=0;

    printf("Enter range for sum of prime numbers : ");
    scanf("%d",&n);
    
    for(i= 2; i<= n; i++)

   {

    f=0;

    for(k=2; k<=i/2; k++)

   {

        if((i % k) == 0){

        f++;

        break;

        }

    }

   if(f==0)

   sum += i;

   }

   printf("\n   Sum of prime numbers is:  %d ",sum);

   return 0;

}

File: solutions/problem-10/sum_of_primes.py
def sum_of_primes(n):
    prime = [True] * (n + 1)
    p = 2
    while p * p <= n:
        if prime[p] == True:
            i = p * 2
            while i <= n:
                prime[i] = False
                i += p
        p += 1   
    sum = 0
    for i in range (2, n + 1):
        if(prime[i]):
            sum += i
    return sum
n = 2000000
print sum_of_primes(n)